File: backend/src/main/scala/microservices/feedback/objects/OpenComplaintManagerThreadPlannerRequest.scala
// OpenComplaintManagerThreadPlannerRequest：feedback 域后端对象定义。
package com.typesafe.travel.feedback.domain
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class OpenComplaintManagerThreadPlannerRequest(
    complaintMessageId: String,
    siteAdminActorId: String
)
object OpenComplaintManagerThreadPlannerRequest:
  given sourceEncoder: Encoder[OpenComplaintManagerThreadPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[OpenComplaintManagerThreadPlannerRequest] = deriveDecoder

File: frontend/src/microservices/feedback/objects/OpenComplaintManagerThreadPlannerRequest.ts
// 本文件定义 feedback 模块的 `OpenComplaintManagerThreadPlannerRequest`，用于把投诉消息转换为站内客服管理线程。

export type OpenComplaintManagerThreadPlannerRequest = {
  complaintMessageId: string
  siteAdminActorId: string
}